src/ucp/api/ucp.h

```diff
@@ -679,10 +682,40 @@ typedef struct ucp_ep_params {
      * Handler to process transport level failure.
      */
     ucp_err_handler_t       err_handler;
+
+    /**
+     * User data associated with an endpoint. See @ref ucp_ep_poll_t.
+     * @a user_data redefines @ref ucp_err_handler_t::arg if set both
```

Comment:
We already have user defined data pointer associated with an endpoint inside `ucp_err_handler_t`. Initially, it was used only for error handling purposes. So, i do not see any use case when user need to have 2 different pointers associated with the same EP.